infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -439,14 +418,12 @@ public void clearLocalStreams() {
         // either on Source or Sink because we would be enforcing an update
         // without opening the stream, hence, leading to "apparent" data loss as
         // checkpoint won't run on these streams
-        Set<UUID> streamsToQuery = streamViewMap.keySet().stream()
+        Set<UUID> streamsToQuery = config.getStreamsIdToNameMap().keySet().stream()
```

Comment:
this map should be refreshed because clear() is done after the transfer phase which can potentially take a long time to complete.